src/Chirp.Web/Pages/Models/ChirpModel.cshtml.cs

```diff
@@ -42,8 +42,9 @@ public Author GetAuthor(string? authorName = null)
         authorName ??= GetUserName();
         if (authorName == null) throw new Exception("User is not authenticated!");
 
-        var author = _repositoryManager.AuthorRepository.FindAuthorsByName(authorName).FirstOrDefault();
-        return author ?? throw new Exception("Author doesn't exist!");
+        var author = RepositoryManager.AuthorRepository.FindAuthorsByName(authorName);
+        var enumerable = author.ToList();
+        return !enumerable.Any() ? RepositoryManager.AuthorRepository.CreateAuthor(new Author{Email = "", Name = authorName}) : enumerable.First();
```

Comment:
I personally disagree with creating new author if not found.

Reply:
Well when do we otherwise create authors? I get what you're saying and I agree that it is the best option, but right now we throw exceptions each time a new user logs into our service.